leases/context.go

```diff
@@ -0,0 +1,24 @@
+package leases
+
+import "context"
+
+type leaseKey struct{}
+
+// WithLease sets a given lease on the context
+func WithLease(ctx context.Context, lid string) context.Context {
+	ctx = context.WithValue(ctx, leaseKey{}, lid)
```

Comment:
What appends if a context already has a lease?  Since the client is creating leases for some actions what will happen if I create a lease and call these methods?

Reply:
I was thinking about this, I might just add a helper function in the client which first checks if there is a lease, and uses it if it already has one, rather than always creating, setting, and deleting it.

Reply:
Maybe just make this function not add a new lease if one is already set?

Reply:
The only problem with that is this is called after `CreateLease`

Reply:
Should CreateLease return the context as well? Maybe lease implements context.Context?

Reply:
I added a commit which adds that helper function I was thinking of. We could change the interface to expose that function as the only way to create leases if it is more user friendly.